### ISSUE TITLE
Use switch instead of button for Power

### ISSUE THREAD

**Issue by felipecrs** [zhbjsh/homeassistant-ssh#36]
Hey, it would be great if instead of exposing a button, you exposed a switch entity instead.

Switch entities have the advantage of showing the status on themselves, so that you can turn it on or off based on the status from a user perspective.

In my case, this would allow me to ditch [WOL](https://www.home-assistant.io/integrations/wake_on_lan/) integration (which exposes a switch) entirely.

And, for me the biggest benefit is that you can expose switch entities to assistants like Google Home, so that I can "Hey Google, turn on computer".

**Comment by felipecrs** [comment]
> Probably because its not always possible to determine whether the device is on or off

Oh. How do you know when the _Power_ button should do WOL or do the shutdown command then? I would say the very same condition is good enough (SSH connected?).

**Comment by felipecrs** [comment]
> If I have time, I will give it a try again, maybe I can make it optional, so users can choose between the buttons and the switch.

Right, I think it makes sense.

**Comment by zhbjsh** [comment]
> > Probably because its not always possible to determine whether the device is on or off
> 
> Oh. How do you know when the _Power_ button should do WOL or do the shutdown command then? I would say the very same condition is good enough (SSH connected?).

Right, probably I didnt like the way the switch behaved when you turn it on and it would immediately go back to the off position until the device finally is online...

**Comment by felipecrs** [comment]
Oh, you can maybe ask the user for that. How much time to wait for the device to come online after sending the magic packet.

**Comment by saberraz** [comment]
It would be indeed a nice improvement to use switch. Currently, Google Home integration with HA does not recognise the buttons.

**Comment by zhbjsh** [comment]
Switch is now finally available in [version 1.3.0](https://github.com/zhbjsh/homeassistant-ssh/releases/tag/v1.3.0)

**Comment by felipecrs** [comment]
Right now, the switch seems to show up as _Unavailable_ when _Allow to turn the device off_ is not enabled.

Should not it be shown as _On_/_Off_, and throw an error if user tries to turn it off without _Allow to turn the device off_?

**Comment by zhbjsh** [comment]
You are right, that would indeed be better. Unfortunately I'm having a strange problem with the switch: If an error is raised, the switch doesn't go back to its initial position automatically, but only after reloading the browser page. This means the switch will show the wrong state in the UI every time an error is raised. So far I couldn't figure out what's causing this issue...

**Comment by felipecrs** [comment]
I'm pretty sure this is a frontend behavior. But maybe you can work around it by updating the state to again to ON after/before throwing the error (even if it never became OFF). I think that may cause the frontend to reload.
